From: Torge Matthies <openglfreak@googlemail.com>
Subject: [PATCH 1/3] powrprof: Add PowerUnregisterSuspendResumeNotification stub.
Message-Id: <20220227010235.276215-1-openglfreak@googlemail.com>
Date: Sun, 27 Feb 2022 02:02:33 +0100

Allegedly needed by Final Fantasy VII Remake to not crash on exit.

Signed-off-by: Torge Matthies <openglfreak@googlemail.com>
---
 dlls/powrprof/powrprof.c    | 7 +++++++
 dlls/powrprof/powrprof.spec | 1 +
 include/powrprof.h          | 1 +
 3 files changed, 9 insertions(+)

diff --git a/dlls/powrprof/powrprof.c b/dlls/powrprof/powrprof.c
index 11111111111..11111111111 100644
--- a/dlls/powrprof/powrprof.c
+++ b/dlls/powrprof/powrprof.c
@@ -333,6 +333,13 @@ DWORD WINAPI PowerEnumerate(HKEY key, const GUID *scheme, const GUID *subgroup,
 DWORD WINAPI PowerRegisterSuspendResumeNotification(DWORD flags, HANDLE recipient, PHPOWERNOTIFY handle)
 {
     FIXME("(0x%08lx,%p,%p) stub!\n", flags, recipient, handle);
+    *handle = (HPOWERNOTIFY)0xdeadbeef;
+    return ERROR_SUCCESS;
+}
+
+DWORD WINAPI PowerUnregisterSuspendResumeNotification(HPOWERNOTIFY handle)
+{
+    FIXME("(%p) stub!\n", handle);
     return ERROR_SUCCESS;
 }
 
diff --git a/dlls/powrprof/powrprof.spec b/dlls/powrprof/powrprof.spec
index 11111111111..11111111111 100644
--- a/dlls/powrprof/powrprof.spec
+++ b/dlls/powrprof/powrprof.spec
@@ -18,6 +18,7 @@
 @ stdcall PowerReadDCValue (ptr ptr ptr ptr ptr ptr ptr)
 @ stdcall PowerReadFriendlyName (ptr ptr ptr ptr ptr ptr)
 @ stdcall PowerRegisterSuspendResumeNotification(long ptr ptr)
+@ stdcall PowerUnregisterSuspendResumeNotification(ptr)
 @ stdcall ReadGlobalPwrPolicy (ptr)
 @ stdcall ReadProcessorPwrScheme (long ptr)
 @ stdcall ReadPwrScheme (long ptr)
diff --git a/include/powrprof.h b/include/powrprof.h
index 11111111111..11111111111 100644
--- a/include/powrprof.h
+++ b/include/powrprof.h
@@ -161,6 +161,7 @@ BOOLEAN WINAPI IsPwrShutdownAllowed(VOID);
 BOOLEAN WINAPI IsPwrSuspendAllowed(VOID);
 DWORD   WINAPI PowerEnumerate(HKEY, const GUID *, const GUID *, POWER_DATA_ACCESSOR, ULONG, UCHAR *, DWORD *);
 DWORD   WINAPI PowerRegisterSuspendResumeNotification(DWORD, HANDLE, PHPOWERNOTIFY);
+DWORD   WINAPI PowerUnregisterSuspendResumeNotification(HPOWERNOTIFY);
 BOOLEAN WINAPI ReadGlobalPwrPolicy(PGLOBAL_POWER_POLICY);
 BOOLEAN WINAPI ReadProcessorPwrScheme(UINT, PMACHINE_PROCESSOR_POWER_POLICY);
 BOOLEAN WINAPI ReadPwrScheme(UINT, PPOWER_POLICY);

-- 
2.35.1

